L2/op_conductor_local_devnet.md

```diff
@@ -0,0 +1,149 @@
+export OP_CONDUCTOR_RPC_ENABLE_ADMIN=true
+export OP_CONDUCTOR_RPC_ENABLE_PROXY=true
+export OP_CONDUCTOR_RPC_PORT=6660
+export OP_CONDUCTOR_PAUSED=false
+export OP_CONDUCTOR_RAFT_BOOTSTRAP=true
+
+# Start op-conductor
+./bin/op-conductor
+```
+5. Configure and Run [op-conductor-ops](https://github.com/ethereum-optimism/infra/tree/main/op-conductor-ops)
+```bash
+# Install dependencies. Follow readme.md to install poetry first
+poetry install
+
+# Configure op-conductor-ops
+cp example.config.toml config.toml
+
+# Example configuration
+[networks.op-network-N]
+sequencers = [
+    "sequencer-0",
+]
+[sequencers.sequencer-0]
+raft_addr = "<sequencer-0-ip-address>:50050"
+conductor_rpc_url = "http://<sequencer-0-ip-address>:6660"
+node_rpc_url = "http://<sequencer-0-ip-address>:8547"
+voting = true
+
+# Check the status of the first sequencer
+./op-conductor-ops status op-network-N
+```
+
+6. Add Additional Sequencers
+ - Deploy two sequencers (op-geth and op-node) on separate machines.
+     - Enable sequencer mode for op-node: OP_NODE_SEQUENCER_ENABLED=true
+     - Use the same sequencer key for op-node: OP_NODE_P2P_SEQUENCER_KEY
+     - Add op-node options:
+        ```bash
+        export OP_NODE_CONDUCTOR_ENABLED=true
+        export OP_NODE_CONDUCTOR_RPC=<conductor-rpc-endpoint>
+        ```
+     - To sync data from other sequencers, we need to add three options accroding to this [doc](https://github.com/ethstorage/pm/blob/main/L2/beta_testnet_new_node.md)
+        ```bash
+         --p2p.static # other two sequencers' address
+         --p2p.no-discovery
+         --p2p.sync.onlyreqtostatic
+         ```
+  - Wait for the new sequencer to start and get synced up with the rest of the nodes
+  - Once the new sequencer is synced up, manually or use automation to add it to the cluster by calling `conductor_addServerAsVoter` json rpc method on the leader sequencer
+      - ```bash
+        curl -X POST -H "Content-Type: application/json" --data \
+            '{"jsonrpc":"2.0","method":"conductor_addServerAsVoter","params":[<raft-server-id>, <raft-consensus-addr>, <raft-config-version>],"id":1}'  \
+            http://127.0.0.1:8547
+        ```
+  - Resume the conductor on the new sequencer by calling `conductor_resume` json rpc method on op-conductor
+      - ```bash
+        curl -X POST -H "Content-Type: application/json" --data \
+            '{"jsonrpc":"2.0","method":"conductor_resume","params":[],"id":1}'  \
+            http://127.0.0.1:8547
+        ```
+
+7. Update op-conductor-ops Configuration
+```bash
+# install poetry and run "poetry install" first
+
+cp example.config.toml config.toml
+
+# config the op-network-N in config.toml
+[networks.op-network-N]
+sequencers = [
+    "sequencer-0",
+    "sequencer-1",
+    "sequencer-2",
+]
+[sequencers.sequencer-0]
+raft_addr = "<sequencer-0-ip-address>:50050"
+conductor_rpc_url = "http://<sequencer-0-ip-address>:6660"
+node_rpc_url = "http://<sequencer-0-ip-address>:8547"
+voting = true
```

Comment:
Is the configuration doing the same thing as `conductor_addServerAsVoter` above?

Reply:
The command `update_cluster_membership` will help to call `conductor_addServerAsVoter` according to the `voting` setting. You can refer to [here](https://github.com/ethereum-optimism/infra/blob/cf71fde505979111a922e8f9f75e7cacf6b4b503/op-conductor-ops/op-conductor-ops.py#L296)

Reply:
Incorporate the above information in this [commit](https://github.com/ethstorage/pm/pull/178/commits/ff569c8c001bbbf974ff6d97d3cb584ceed1c396)